### ISSUE TITLE
Scrum Report 2023-06-25

### ISSUE THREAD

**Issue by wgervasio** [leoxyu/455-project#31]
Describe (for the project): 

> csid: 
> 
> (1) what you’ve worked on in the previous two weeks, 
> 
> (2) what you’ll be working on in the next two weeks, and 
> 
> (3) where you might have gotten stuck during the last two weeks.

These should be quite short (a sentence for each of the three items above).

**Comment by wgervasio** [comment]
csid: s0m3b

(1) what you’ve worked on in the previous two weeks,
UI components for search page; searching functionality. Also ended up making a last minute playlist list page. Fixed server endpoints and redux reducer to add playlists, add songs, and to delete playlists. Half finished the edit playlist details functionality.

(2) what you’ll be working on in the next two weeks, and
Finish the searching functionality. Combining the filtering functionality. Finishing the UI for the search page. Work on recommendation system 

(3) where you might have gotten stuck during the last two weeks.
Blocked on the spotify searching and getting a reasonable youtube search going without all our dev credits being wasted due to the API. I found some alternatives that I can go and work with,

**Comment by AlexZhou876** [comment]
csid: n4z2b

(1) what you’ve worked on in the previous two weeks,
redux thunks and reducers for playlists and songs, styling for playlist list component, PR reviews and discussions

(2) what you’ll be working on in the next two weeks, and
helping create mongoDB schema, creating additional REST endpoints if needed, helping with finalizing frontend components, particularly playlist

(3) where you might have gotten stuck during the last two weeks.
N/A

**Comment by leoxyu** [comment]
csid: z5z2b

(1) what you’ve worked on in the previous two weeks,
Redux and APIs for login and register. Also added navbar and helped with debugging OAuth

(2) what you’ll be working on in the next two weeks, and
Implementing mongoDB with login/register. Helping with finalizing UI elements and adding

(3) where you might have gotten stuck during the last two weeks.
N/A

**Comment by moseshsu01** [comment]
csid: t8v2b

(1) what you’ve worked on in the previous two weeks
Tried spotify web playback sdk, found out it wasn't what we wanted, continued implementing spotify iframe api & got through some hurdles with that, then I made the UI for the audio player component & the functionality for that

(2) what you’ll be working on in the next two weeks, and
Finish the UI for the player (adding song title and image + proper positioning between pages), hook it up so it can receive songs/playlists + probably need redux for that

(3) where you might have gotten stuck during the last two weeks.
Spotify iframe api is hard to work with but I eventually got around where I got stuck

**Comment by Ning114** [comment]
csid: c9x2b

(1) what you’ve worked on in the previous two weeks

I worked on getting the OAuthentication for spotify's web API working such that our client can fetch an access key and use it to obtain more information from spotify's API

(2) what you’ll be working on in the next two weeks, and

I am nearly complete the OAuth, and expect to finish that alongside all other spotify API related backend features by the start of the next sprint.

(3) where you might have gotten stuck during the last two weeks.

I had to spend quite alot of time researching and understanding the flow of the OAuth of Spotify's API. However by now I believe I have attained a fair understanding and can move onto implementation
